.github/workflows/build.yml

```diff
@@ -468,19 +467,36 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push Docker image
+      - name: Build and push Docker image (official release)
         id: docker_build
+        if: '!github.event.release.prerelease'
         uses: docker/build-push-action@v3
         with:
           push: true
           provenance: false
           tags: ghcr.io/sage-bionetworks/synapsepythonclient:latest,ghcr.io/sage-bionetworks/synapsepythonclient:${{ env.RELEASE_VERSION }}
           file: ./Dockerfile
           platforms: linux/amd64
-          cache-from: type=registry,ref=ghcr.io/sage-bionetworks/synapsepythonclient:latest
-          cache-to: type=inline
-      - name: Output image digest
-        run: echo "The image digest is ${{ steps.docker_build.outputs.digest }}"
+          cache-from: type=registry,ref=ghcr.io/sage-bionetworks/synapsepythonclient:build-cache
+          cache-to: type=registry,mode=max,ref=ghcr.io/sage-bionetworks/synapsepythonclient:build-cache
+      - name: Build and push Docker image (pre-release)
+        id: docker_build_prerelease
+        if: 'github.event.release.prerelease'
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          provenance: false
+          tags: ghcr.io/sage-bionetworks/synapsepythonclient:${{ env.RELEASE_VERSION }}-prerelease
+          file: ./Dockerfile
+          platforms: linux/amd64
+          cache-from: type=registry,ref=ghcr.io/sage-bionetworks/synapsepythonclient:build-cache-prerelease
```

Comment:
New caches `build-cache` and `build-cache-prerelease` will now store the Docker caches so that the caches between official release and pre-release are separate.